packages/matrix/tests/registration-with-token.spec.ts

```diff
@@ -18,6 +18,16 @@ import { registerUser, createRegistrationToken } from '../docker/synapse';
 
 const REGISTRATION_TOKEN = 'abc123';
 
+// TODO after we start creating realms as part of user creation we'll need to
+// figure out how to perform better isolation for these tests. Creating a user
+// will result in DB and filesystem state changes that will bleed into the other
+// matrix tests. One idea is that since playwright is running on the server we
+// could perform realm server isolation by using randomly generated DB names's
+// like we do in the realm server tests along with resetting the underlying
+// filesystem and restarting the realm server in between tests that we expect to
+// mutate realm state. These would be rather expensive tests so we should be
+// prudent around the realm server isolation.
```

Comment:
Maybe we could introduce a virtual (in-memory) filesystem for tests

Reply:
There will still be DB artifacts though that need to be isolated regardless